template/src/lib/service.broker.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Service broker and start/init function.
+ * Use it everywhere (production, dev, tests).
+ * Adapt using proper config (env vars).
+ *
+ * Copyright Byte Technology 2020. All rights reserved.
+ */
+import { TypedServiceBroker } from 'moleculer-service-ts';
+import { Service, Middleware, ServiceBroker, EventSchema } from 'moleculer';
+import {
+  MikroConnector,
+  DatabaseContextManager,
+  MoleculerMikroContext
+} from 'moleculer-context-db';
+import { ServiceAction, ServiceEvent, ServiceName } from '../service.types';
+
+import { brokerConfig } from './moleculer/broker.config';
+import { {{capitalizedServiceName}}Service } from '../{{serviceName}}.service';
+import { dbConnector, initDbConnector } from '../db.connector';
+
+let service: Service;
+let started = false;
+
+export const broker: TypedServiceBroker<
+  ServiceAction,
+  ServiceEvent,
+  ServiceName
+> = new TypedServiceBroker(brokerConfig);
+
+const LogMiddleware: Middleware = {
+  localAction(next: any) {
+    return function logAction(this: ServiceBroker, ctx: MoleculerMikroContext) {
+      broker.logger.info(
+        `Action '${(ctx.action as any).name}'.  '${ctx.caller}'. CallerNode '${
+          ctx.nodeID
+        }'.`
+      );
+      return next(ctx);
+    };
+  },
+  localEvent(next: any, event: EventSchema) {
+    return function logEvent(payload: any, sender: string, eventName: string) {
+      broker.logger.info(`Event ${event.name}. From service '${sender}'`);
+      return next(payload, sender, eventName);
+    };
+  }
+};
+
+export async function startService(): Promise<Service> {
+  if (started) {
+    throw new Error(
+      `startService() already called. Now you can only call stopService().`
+    );
+  }
+  started = true;
+
+  broker.middlewares.add(LogMiddleware);
+
+  await initDbConnector();
```

Comment:
This still assumes that all services need a dbConnector(). Some services such ```api``` will never talk to a DB directly. They will not need a ```dbConnector```.